assets/js/app/Data.js

```diff
@@ -175,30 +178,94 @@ LocusZoom.Data.AssociationSource.prototype.parseResponse = function(resp, chain,
         }
         records.push(record);
     }
-    var res = {header: chain.header || {}, body: records};
-    return res;
+    return records;
+};
+
+LocusZoom.Data.Source.prototype.parseObjectsToObjects = function(x, fields, outnames, trans) {
+    //intended for an array of objects
+    // [ {"id":1, "val":5}, {"id":2, "val":10}]
+    var records = [];
+    var fieldFound = [];
+    for (var i = 0; i < x.length; i++) {
+        var record = {};
+        for (var j=0; j<fields.length; j++) {
+            var val = x[i][fields[j]];
+            if (typeof val != "undefined") {
```

Comment:
It looks like `fieldFound` is only getting an index populated with 1 if the field at that index `j` is found. Thus if I've got a list of four fields: `["a", "b", "c", "d"]` and "c" isn't found, `fieldFound` would be `[0, 1, 3]`. At the end of the function in the `forEach` on line 203, then, `v` would always be "1" and an exception would never be thrown for a missing field. Should this conditional have an `else` that sets `fieldFound[j]` to `false` or `0` to get the desired effect?